frontend/packages/core/src/Network/index.ts

```diff
@@ -0,0 +1,66 @@
+import type { AxiosError } from "axios";
+import axios from "axios";
+
+import type { ClientError, ClutchError, NetworkError } from "./errors";
+import clutchError from "./errors";
+
+/**
+ * HTTP response status.
+ *
+ * Responses are grouped in five classes:
+ *  - Informational responses (100–199)
+ *  - Successful responses (200–299)
+ *  - Redirects (300–399)
+ *  - Client errors (400–499)
+ *  - Server errors (500–599)
+ */
+export interface HttpStatus {
+  /** The status code. */
+  code: number;
+  /** The status message. */
+  text: string;
+}
+
+const errorInterceptor = (error: AxiosError): Promise<ClutchError | NetworkError | ClientError> => {
```

Comment:
curious, can we standardize on ClutchError as return type?

e.g. instead of ClientError just make a fake clutch error that says i'm a client error? or containing a details field that lets us know for certain it's a client error?

seems like it would simplify handling as the error bubbles up through the rest of the code.